### DISCUSSION TITLE
Graph construction problem: a combination of multigraph and heterogeneous graph

### DISCUSSION THREAD

**Discussion by RENEK-bool** [pyg-team/pytorch_geometric#9010] [Q&A] [answered]
Hello, I am currently trying to build a graph with two types of edges, and sometimes there may be two edges between a pair of nodes, I wonder whether pytorch geometric supports such modeling, and is there any suitable model to train? This seems to be a combination of multigraph and heterogeneous graph.

**Comment by rusty1s** [comment, marked as answer]
PyG supports both multi-graphs and heterogeneous graphs. If you have duplicated edges, message passing would send the information from a neighbor twice.
